### PR TITLE
fix service port and ingress listener port merge

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

ingress listener port is the target port, so should compare with service targetport

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@howardjohn

File: pilot/pkg/networking/core/v1alpha3/listener_test.go [hunk 1]
```diff
@@ -1482,7 +1502,7 @@ func testInboundListenerConfigWithSidecarIngressPortMergeServicePort(t *testing.
 		Configs:  []config.Config{sidecarConfig},
 	}, proxy)
 	l := xdstest.ExtractListener(model.VirtualInboundListenerName, listeners)
-	if len(l.FilterChains) != 12 {
+	if len(l.FilterChains) != 14 {
 		t.Fatalf("expected %d listener filter chains, found %d", 12, len(l.FilterChains))
```
Comment (ymesika) [inline pilot/pkg/networking/core/v1alpha3/listener_test.go:1506]: 12 -> 14 here too

**Review by ymesika — CHANGES_REQUESTED** [review]
Looks good beside the comment I left

File: pilot/pkg/networking/core/v1alpha3/listener_inbound.go [hunk 2]
```diff
@@ -412,7 +412,8 @@ func (lb *ListenerBuilder) getFilterChainsByServicePort(chainsByPort map[uint32]
 		}
 		actualWildcards, _ := getWildcardsAndLocalHost(lb.node.GetIPMode())
 		if enableSidecarServiceInboundListenerMerge && sidecarScope.HasIngressListener() &&
-			ingressPortListSet.Contains(int(port.Port)) {
+			// ingress listener port means the target port, may not equal to service port
+			ingressPortListSet.Contains(int(port.TargetPort)) {
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener_inbound.go:416]: I am a bit confused, Port and TargetPort are always the same, right? So is this just making it more clear, or fixing a big?

File: pilot/pkg/networking/core/v1alpha3/listener_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener_test.go:1506]: good catch

File: pilot/pkg/networking/core/v1alpha3/listener_inbound.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener_inbound.go:416]: This is fixing a bug, as you can see in L407~412. The port and target port are for service concept, can be different

```
		port := ServiceInstancePort{
			Name:       i.ServicePort.Name,
			Port:       uint32(i.ServicePort.Port),
			TargetPort: i.Endpoint.EndpointPort,
			Protocol:   i.ServicePort.Protocol,
		}```

**Review by howardjohn — APPROVED** [review]

**Pull request merged**